### PR TITLE
GP2 email validation

### PR DESCRIPTION
**Pull request by markcsims**
- updates for now validation expressions
- adds telephone and email validation to gp2 user form
- renames variable
- adds telephone and email validation to gp2 server

### CODE REVIEW

File: packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx [hunk 1]
```diff
@@ -125,4 +125,79 @@ describe('ContactInformationModal', () => {
...
+    renderContactInformation({
+      email,
+      secondaryEmail: '',
+      telephone: undefined,
+      onSave,
+    });
+
+    userEvent.type(
+      screen.getByRole('textbox', {
+        name: 'Alternative Email (optional) An alternative way for members to contact you. This will not affect the way that you login.',
```
Comment (jhonny-m) [inline packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx:141]: you can use `/alternative email/i`

File: packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx [hunk 2]
```diff
@@ -125,4 +125,79 @@ describe('ContactInformationModal', () => {
...
+    renderContactInformation({
+      email,
+      secondaryEmail: '',
+      telephone: undefined,
+      onSave,
+    });
+
+    userEvent.type(
+      screen.getByRole('textbox', {
+        name: 'Telephone Number (optional) Please note: this will only be visible to admins.',
```
Comment (jhonny-m) [inline packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx:191]: `/telephone number/i`

File: packages/validation/src/expressions.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// remember to update link patterns in the squidex schema
+
+export const urlExpression =
+  "^(?:http(s)?:\\/\\/)[\\w.-]+(?:\\.[\\w\\.-]+)+[\\w\\-\\._~:\\/?#%[\\]@!\\$&'\\(\\)\\*\\+,;=.]+$";
+
+export const emailExpression =
+  "^[a-zA-Z0-9.!#$%&’*+\\/=?^_`'{|}~-]+@[a-zA-Z0-9-]+(?:.[a-zA-Z0-9-]+)*$";
+
+export const telephoneNumberExpression =
+  '^\\(?\\d{3}\\)?[\\s.-]?\\d{3}[\\s.-]?\\d{3,4}$';
```
Comment (jhonny-m) [inline packages/validation/src/expressions.ts:12]: did you learn them or got them from chatgtp?

**Review by jhonny-m — APPROVED** [review]

File: packages/validation/src/expressions.ts (hunk 3 above)
Comment (markcsims) [reply packages/validation/src/expressions.ts:12]: better - squidex

Comment (markcsims) [reply packages/validation/src/expressions.ts:12]: They are the same as we use for validating a field at cms

File: packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx (hunk 1 above)
Comment (markcsims) [reply packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx]: done

File: packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx (hunk 2 above)
Comment (markcsims) [reply packages/gp2-components/src/organisms/__tests__/ContactInformationModal.test.tsx]: done

**Pull request merged**